core/chains/evm/client/chain_client.go

```diff
@@ -181,96 +200,140 @@ func (c *chainClient) BlockByHash(ctx context.Context, hash common.Hash) (b *typ
 
 // TODO-1663: return custom Block type instead of geth's once client.go is deprecated.
 func (c *chainClient) BlockByNumber(ctx context.Context, number *big.Int) (b *types.Block, err error) {
-	rpc, err := c.multiNode.SelectNodeRPC()
+	rpc, err := c.multiNode.SelectRPC()
 	if err != nil {
 		return b, err
 	}
 	return rpc.BlockByNumberGeth(ctx, number)
 }
 
 func (c *chainClient) CallContext(ctx context.Context, result interface{}, method string, args ...interface{}) error {
-	return c.multiNode.CallContext(ctx, result, method, args...)
+	rpc, err := c.multiNode.SelectRPC()
+	if err != nil {
+		return err
+	}
+	return rpc.CallContext(ctx, result, method, args...)
 }
 
 func (c *chainClient) CallContract(ctx context.Context, msg ethereum.CallMsg, blockNumber *big.Int) ([]byte, error) {
-	return c.multiNode.CallContract(ctx, msg, blockNumber)
+	rpc, err := c.multiNode.SelectRPC()
+	if err != nil {
+		return nil, err
+	}
+	return rpc.CallContract(ctx, msg, blockNumber)
 }
 
 func (c *chainClient) PendingCallContract(ctx context.Context, msg ethereum.CallMsg) ([]byte, error) {
-	return c.multiNode.PendingCallContract(ctx, msg)
+	rpc, err := c.multiNode.SelectRPC()
+	if err != nil {
+		return nil, err
+	}
+	return rpc.PendingCallContract(ctx, msg)
 }
 
 // TODO-1663: change this to actual ChainID() call once client.go is deprecated.
 func (c *chainClient) ChainID() (*big.Int, error) {
-	//return c.multiNode.ChainID(ctx), nil
-	return c.multiNode.ConfiguredChainID(), nil
+	rpc, err := c.multiNode.SelectRPC()
+	if err != nil {
+		return nil, err
+	}
+	// TODO: Progagate context
+	return rpc.ChainID(context.Background())
```

Comment:
Hm, not sure that it worth changing behaviour here. We may return configured chainID.
Also could you remove the TODO comment. It seems to be outdated.